pkg/proxy/iptables/proxier.go

```diff
@@ -1276,13 +1368,24 @@ func (proxier *Proxier) syncProxyRules() {
 	natLines := append(natChains.Bytes(), natRules.Bytes()...)
 	lines := append(filterLines, natLines...)
 
-	glog.V(3).Infof("Restoring iptables rules: %s", lines)
-	err = proxier.iptables.RestoreAll(lines, utiliptables.NoFlushTables, utiliptables.RestoreCounters)
+	filterSame, err := sameIptablesRuleset(utiliptables.TableFilter, filterLines, iptablesFilterSaveRaw)
 	if err != nil {
-		glog.Errorf("Failed to execute iptables-restore: %v", err)
-		// Revert new local ports.
-		revertPorts(replacementPortsMap, proxier.portsMap)
-		return
+		glog.Warningf("failed to parse iptables filter rules: %v", err)
```

Comment:
In some failure cases (eg, iptables-save output format changes), every single call to sameIptablesRuleset() will fail... we might want to not log every time?